tests/console/osc.pm

```diff
@@ -0,0 +1,81 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+# Author: Gustavo Figueira <gfigueira@suse.com>
+
+use base "consoletest";
+use testapi;
+use utils;
+use Utils::Backends 'use_ssh_serial_console';
+use version_utils qw(is_sle get_os_release is_tumbleweed);
+use strict;
+use warnings;
+
+sub repos_and_pkgs_sle {
+    my ($hub_version, $arch) = @_;
+    my $repo_version = $hub_version;
+    $repo_version =~ s/\./_SP/;
+    assert_script_run("SUSEConnect --product PackageHub/" . $hub_version . "/" . $arch);
+    zypper_call("ar http://download.suse.de/ibs/SUSE:/CA/SLE_" . $repo_version . "/ suseca");
+    zypper_call("in osc obs-service-source_validator ca-certificates-suse");
+}
+
+sub prepare_oscrc {
+    my ($oscrc_path, $IBSUSER, $SECRET) = @_;
+    assert_script_run("wget --quiet " . data_url('oscrc') . " -O " . $oscrc_path, timeout => 30);
+    assert_script_run("perl -pi -e 's/TAGIBSUSER/" . $IBSUSER . "/g' " . $oscrc_path);
+    assert_script_run("perl -pi -e 's/TAGSECRET/" . $SECRET . "/g' " . $oscrc_path);
+}
+
+sub co_and_build {
+    my ($codestream, $testpkg) = @_;
+    my $api_opensuse = "https://api.opensuse.org";
+
+    if (is_sle) {
+        assert_script_run("yes 1 | osc co " . $codestream . " " . $testpkg);
+    } else {
+        assert_script_run("yes 1 | osc -A " . $api_opensuse . " co " .  $codestream . " " . $testpkg);
+    }
```

Comment:
```suggestion
    assert_script_run("yes 1 | osc " . (is_sle ? '' : "-A " . $api_opensuse ) . " co " .  $codestream . " " . $testpkg);
```

Reply:
I have no problem with the ternary operator, especially if it concerns branching of simple assignment expressions, but I believe this change only impairs the readability of the code.

Reply:
How about:

```suggestion
    my $cmd = (is_sle ? '' : "-A $api_opensuse ") . co $codestream $testpkg");
    assert_script_run("yes 1 | osc $cmd");
```

But if you want to avoid the ternary operator one can also do

```suggestion
    my $cmd = "osc co $codestream $testpkg";
    $cmd = "osc -A " . $api_opensuse $cmd" unless is_sle;
    assert_script_run "yes 1 | $cmd";
```

The important part is to reduce the duplication so that relevant common parts are better separated from important differences

Reply:
I strongly suggest that the ternary operator be avoided, given that this code base is maintained by people with varying levels of Perl proficiency.

Reply:
Regardless of the level of proficiency we should avoid duplication. If you prefer you can take the "if"-way which I also provided. But I suggest to learn how ternary operators work in *all* languages. This is not a perl-specific thing.